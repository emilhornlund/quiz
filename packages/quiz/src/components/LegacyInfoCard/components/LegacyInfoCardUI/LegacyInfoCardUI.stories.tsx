import type { Meta, StoryObj } from '@storybook/react'

import LegacyInfoCardUI from './LegacyInfoCardUI'

const meta = {
  title: 'Surfaces/LegacyInfoCard',
  component: LegacyInfoCardUI,
} satisfies Meta<typeof LegacyInfoCardUI>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    migrated: false,
    legacyDomain: 'quiz.emilhornlund.com',
    targetDomain: 'klurigo.com',
    onDismiss: () => undefined,
  },
} satisfies Story

export const MigratedButUserNotAuthenticated = {
  args: {
    migrated: true,
    legacyDomain: 'quiz.emilhornlund.com',
    targetDomain: 'klurigo.com',
    onDismiss: () => undefined,
  },
} satisfies Story
