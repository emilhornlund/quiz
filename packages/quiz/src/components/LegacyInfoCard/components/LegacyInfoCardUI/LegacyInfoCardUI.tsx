import React, { FC } from 'react'

import Card from '../../../Card'

export interface LegacyInfoCardUIProps {
  migrated: boolean
  legacyDomain?: string
  targetDomain?: string
  onDismiss: () => void
}

const LegacyInfoCardUI: FC<LegacyInfoCardUIProps> = ({
  migrated,
  legacyDomain,
  targetDomain,
  onDismiss,
}) => (
  <Card
    kind={migrated ? 'success' : 'call-to-action'}
    size="large"
    onDismiss={onDismiss}
    center>
    <span>
      {migrated ? (
        <>
          Your migration’s all set! Pop in or make a fresh account, and we’ll
          take it from here.
        </>
      ) : (
        <>
          Hey quiz champion! Did your brainy adventures happen on{' '}
          <a href={`https://${legacyDomain}`}>{legacyDomain}</a>? No worries— we
          can save all your quiz creations and epic game stats! Just pop back to{' '}
          <a href={`https://${legacyDomain}`}>{legacyDomain}</a> one last time,
          and we’ll magically teleport everything over to{' '}
          <b>
            <i>{targetDomain}</i>
          </b>
          . Ready to blast off your data?
        </>
      )}
    </span>
  </Card>
)

export default LegacyInfoCardUI
