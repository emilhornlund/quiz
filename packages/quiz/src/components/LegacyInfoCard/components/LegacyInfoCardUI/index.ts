export type { LegacyInfoCardUIProps } from './LegacyInfoCardUI'
export { default } from './LegacyInfoCardUI'
