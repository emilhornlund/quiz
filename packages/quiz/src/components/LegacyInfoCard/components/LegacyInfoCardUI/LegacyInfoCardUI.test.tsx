import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import LegacyInfoCardUI from './LegacyInfoCardUI'

describe('LegacyInfoCardUI', () => {
  it('should render a LegacyInfoCardUI as not migrated', async () => {
    const { container } = render(
      <LegacyInfoCardUI
        migrated={false}
        legacyDomain="https://legacy.com"
        targetDomain="https://target.com"
        onDismiss={() => undefined}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render a LegacyInfoCardUI as migrated', async () => {
    const { container } = render(
      <LegacyInfoCardUI
        migrated={true}
        legacyDomain="https://legacy.com"
        targetDomain="https://target.com"
        onDismiss={() => undefined}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
