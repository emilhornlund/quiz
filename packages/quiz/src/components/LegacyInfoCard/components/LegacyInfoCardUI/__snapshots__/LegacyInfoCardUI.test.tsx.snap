// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LegacyInfoCardUI > should render a LegacyInfoCardUI as migrated 1`] = `
<div>
  <div
    class="card kindSuccess sizeLarge center"
  >
    <button
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-xmark"
        data-icon="xmark"
        data-prefix="fas"
        role="img"
        viewBox="0 0 384 512"
      >
        <path
          d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="content"
    >
      <span>
        Your migration’s all set! Pop in or make a fresh account, and we’ll take it from here.
      </span>
    </div>
  </div>
</div>
`;

exports[`LegacyInfoCardUI > should render a LegacyInfoCardUI as not migrated 1`] = `
<div>
  <div
    class="card kindCallToAction sizeLarge center"
  >
    <button
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-xmark"
        data-icon="xmark"
        data-prefix="fas"
        role="img"
        viewBox="0 0 384 512"
      >
        <path
          d="M55.1 73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L147.2 256 9.9 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192.5 301.3 329.9 438.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.8 256 375.1 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192.5 210.7 55.1 73.4z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="content"
    >
      <span>
        Hey quiz champion! Did your brainy adventures happen on
         
        <a
          href="https://https://legacy.com"
        >
          https://legacy.com
        </a>
        ? No worries— we can save all your quiz creations and epic game stats! Just pop back to
         
        <a
          href="https://https://legacy.com"
        >
          https://legacy.com
        </a>
         one last time, and we’ll magically teleport everything over to
         
        <b>
          <i>
            https://target.com
          </i>
        </b>
        . Ready to blast off your data?
      </span>
    </div>
  </div>
</div>
`;
